src/main/java/com/example/onlinebookstore/repository/specifications/PriceSpecificationProvider.java

```diff
@@ -0,0 +1,23 @@
+package com.example.onlinebookstore.repository.specifications;
+
+import com.example.onlinebookstore.model.Book;
+import com.example.onlinebookstore.repository.SpecificationProvider;
+import java.util.Arrays;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PriceSpecificationProvider implements SpecificationProvider<Book> {
+    private static final String PRICE = "price";
+
+    @Override
+    public String getKey() {
+        return PRICE;
+    }
+
+    @Override
+    public Specification<Book> getSpecification(String[] params) {
+        return (root, query, criteriaBuilder) ->
+                root.get(PRICE).in(Arrays.stream(params).toArray());
```

Comment:
It is impossible to use - how do you imagine this price array?

Reply:
Like this
<img width="1044" alt="Screenshot 2023-11-03 at 6 50 14 AM" src="https://github.com/PavloSliepchenko/Online-Book-Store/assets/115042054/23006656-c7c3-4c21-8e55-ce96a3d47a70">